angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -2,163 +2,194 @@
+	app.controller('OptionLinkController', ['$scope', 'layoutService', function($scope, layoutService){
 
-        var configInit = function(){
-            if(!$scope.config) {
-                //setting up defaults since config doesn't exist
-                $scope.config = {
-                    singleElement : false,
-                    arrayName : 'array',
-                    value : 'value',
-                    display : 'display'
-                };
-            }
-        };
+		var configInit = function(){
+			if(!$scope.config) {
+				//setting up defaults since config doesn't exist
+				$scope.config = {
+						singleElement : false,
+						arrayName : 'array',
+						value : 'value',
+						display : 'display'
+				};
+			}
+		};
 
-        var populateWidgetContent = function() {
-            if($scope.portlet.widgetURL && $scope.portlet.widgetType) {
-                //fetch portlet widget json
-                $scope.portlet.widgetData = [];
+		var populateWidgetContent = function() {
+			if($scope.portlet.widgetURL && $scope.portlet.widgetType) {
+				//fetch portlet widget json
+				$scope.portlet.widgetData = [];
 
-                layoutService.getWidgetJson($scope.portlet).then(function(data) {
-                    if(data) {
-                        console.log(data);
-                        //set init
-                        if($scope.config.singleElement) {
-                            //set the default selected url
-                            $scope.portlet.selectedUrl = $scope.portlet.widgetData[$scope.config.value];
-                        } else if($scope.portlet.widgetData[$scope.config.arrayName] && $scope.portlet.widgetData[$scope.config.arrayName].length > 0) {
-                            $scope.portlet.selectedUrl = $scope.portlet.widgetData[$scope.config.arrayName][0][$scope.config.value];
-                        }
-                    } else {
-                        console.warn("Got nothing back from widget fetch for " + $scope.portlet.fname);
-                    }
-                });
-            }
-        };
-        configInit();
-        populateWidgetContent();
-    }]);
+				layoutService.getWidgetJson($scope.portlet).then(function(data) {
+					if(data) {
+						console.log(data);
+						//set init
+						if($scope.config.singleElement) {
+							//set the default selected url
+							$scope.portlet.selectedUrl = $scope.portlet.widgetData[$scope.config.value];
+						} else if($scope.portlet.widgetData[$scope.config.arrayName] && $scope.portlet.widgetData[$scope.config.arrayName].length > 0) {
+							$scope.portlet.selectedUrl = $scope.portlet.widgetData[$scope.config.arrayName][0][$scope.config.value];
+						}
+					} else {
+						console.warn("Got nothing back from widget fetch for " + $scope.portlet.fname);
+					}
+				});
+			}
+		};
+		configInit();
+		populateWidgetContent();
+	}]);
 
```

Comment:
It would look like your text editor (probably eclipse?) is inserting tabs for spaces.  Not a big deal, but when you get a chance, go ahead and change that to insert spaces for tabs.  If you want, I'll show you what I'm talking about and where to do it.